java-api-core/src/com/kinvey/java/store/requests/data/save/SaveListRequest.java

```diff
@@ -52,10 +52,18 @@ public SaveListRequest(ICache<T> cache, NetworkManager<T> networkManager, WriteP
     @Override
     public List<T> execute() throws IOException {
         List<T> ret = new ArrayList<T>();
-        for (T obj : objects){
-            SaveRequest<T> save = new SaveRequest<T>(
-                    cache, networkManager , writePolicy, obj, syncManager);
-            ret.add(save.execute());
+        if (writePolicy == WritePolicy.FORCE_LOCAL) {
+            ret = cache.save(objects);
+            for (T obj : ret){
+                syncManager.enqueueRequest(networkManager.getCollectionName(),
+                        networkManager.saveBlocking(obj));
+            }
+        } else {
+            for (T obj : objects){
+                SaveRequest<T> save = new SaveRequest<T>(
+                        cache, networkManager , writePolicy, obj, syncManager);
+                ret.add(save.execute());
+            }
```

Comment:
What happens in the `LOCAL_THEN_NETWORK` case? Does this also make a bulk save of entities to Realm?

Reply:
In case when write policy is LOCAL_THEN_NETWORK a bulk save won't work. Because there are a bit different logic in LOCAL_THEN_NETWORK case and customer asked only about StoreType.SYNC. I think we can think about improvement this logic for LOCAL_THEN_NETWORK, but I think it should be a different ticket.